### PR TITLE
boards/stm32f4: Remove incorrect UART DMA config from stm32f4 boards

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

Currently a lot of boards do not have DMA streams configured, but do have snippets to enable DMA config for the UART (and sometimes SPI). This configuration is unfortunately incorrect due to this.

This PR sets the DMA configuration for the UART peripherals on the
stm32f4 boards to the undef value to disable DMA usage with UART. This to prevent
from accidentally configuring the DMA to stream 0 channel 0 when enabling DMA.

### Testing procedure

As there is no valid DMA peripheral config available for these boards, this only modifies dead code. 

### Issues/PRs references

I want to add valid DMA config to these boards later on, this is a first PR to ensure that I don't break the current UART config when enabling DMA.

### CODE REVIEW

**Comment by bergzand** [comment]
I've added a commit that also removes the obsolete UART DMA ISR defines.

**Comment by aabadie** [comment]
Well, I don't agree with this PR: the DMA configurations (streams, channels and ISR) are most of the time valid in regard of the DMA implementation on STM32. Maybe some are just copy-pasted, but I'm pretty sure I tested DMA on several F4 boards.

**Comment by bergzand** [comment]
Maybe the channels configured here are valid, but at this moment, the `uart_conf_t` doesn't have a [`dma_stream`](https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32_common/include/periph_cpu_common.h#L455-L486) member. So in no way will the current config compile when DMA (and `UART_USE_DMA` is defined). Furthermore, the removed `UART_0_DMA_ISR` appear only in this config.

**Comment by aabadie** [comment]
There's indeed something wrong in the current code. The `dma` attribute in the common uart configuration struct should be called `dma_stream`, to match what is in the board configurations.

I agree that the ISR defines are unused.

**Comment by aabadie** [comment]
> The dma attribute in the common uart configuration struct should be called dma_stream, to match what is in the board configurations.

it's the `dma` that is used in the uart implementation, so ok to change the boards configurations. But you still have to keep `dma_chan`:

https://github.com/RIOT-OS/RIOT/blob/f81dda2174cfc48febcd538ee4457ed433c69c8c/cpu/stm32_common/periph/uart.c#L349

**Comment by aabadie** [comment]
And there might be a reason to keep the ISR defines. Maybe @vincent-d could tell. Or we can find this info in the related PRs (maybe #9171).

**Comment by aabadie** [comment]
Ok, now I understand.

I would do it in a much more simpler way: none of these boards provide the `periph_dma`, so the build system will prevent DMA builds anyway. So I would go for just removing anything related to DMA (the attributes in the config + the ISR defines).

We can later add DMA related configurations for these boards if needed.

**Comment by bergzand** [comment]
I'm fine with removing the config completely. I'm also fine with splitting the commits into removing the config completely and in a second commit add the `DMA_STREAM_UNDEF` lines.

I plan on a next PR to add config to the relevant DMA config and enable SPI DMA support. The `DMA_STREAM_UNDEF` lines are required to enable the `periph_dma` feature without having to include UART dma support. UART DMA support can be added in a later stage, but I'd rather not enable SPI and UART DMA at the same time to keep things simple.

**Comment by bergzand** [comment]
I mostly want to add UART DMA support in a later stage because it caused issues with a number of tests in the past. So I was thinking of first adding SPI DMA support, leave UART DMA disabled, and later add UART DMA support.

File: boards/f4vi1/include/periph_conf.h [hunk 1]
```diff
@@ -69,15 +69,13 @@ static const uart_conf_t uart_config[] = {
         .tx_af      = GPIO_AF8,
         .bus        = APB2,
         .irqn       = USART6_IRQn,
-#ifdef UART_USE_DMA
-        .dma_stream = 14,
-        .dma_chan   = 5
+#ifdef MODULE_PERIPH_DMA
+        .dma        = DMA_STREAM_UNDEF,
```
Comment (aabadie) [inline boards/f4vi1/include/periph_conf.h:73]:
```suggestion
        .dma        = DMA_STREAM_UNDEF,
        .dma_chan   = 0,  /* or maybe .dma_chan = UINT8_MAX, ? */
```
here and below ?

Comment (bergzand) [reply boards/f4vi1/include/periph_conf.h:73]: Added `UINT8_MAX` here and everywhere

File: boards/stm32f4discovery/include/periph_conf.h [hunk 2]
```diff
@@ -84,17 +84,14 @@ static const uart_conf_t uart_config[] = {
         .tx_af      = GPIO_AF7,
         .bus        = APB1,
         .irqn       = USART3_IRQn,
-#ifdef UART_USE_DMA
-        .dma_stream = 3,
-        .dma_chan   = 4
+#ifdef MODULE_PERIPH_DMA
+        .dma        = DMA_STREAM_UNDEF,
```
Comment (aabadie) [inline boards/stm32f4discovery/include/periph_conf.h:88]:
```suggestion
        .dma        = DMA_STREAM_UNDEF,
        .dma_chan   = UINT8_MAX,
```

**Comment by aabadie** [comment]
Please fix my last comment and squash !

File: boards/stm32f4discovery/include/periph_conf.h (hunk 2 above)
Comment (bergzand) [reply boards/stm32f4discovery/include/periph_conf.h:88]: FIxed!

**Comment by bergzand** [comment]
Squashed!

**Comment by bergzand** [comment]
Thanks @aabadie!

**Pull request merged**